src/electionguard/publish.py

```diff
@@ -0,0 +1,78 @@
+from typing import List
+from jsons import set_serializer
+from os import mkdir, path
+from jsons import set_serializer, dump
+
+
+from .ballot_box import CiphertextBallot
+from .decryption_mediator import PlaintextTally
+from .election import CiphertextElectionContext, ElectionConstants, ElectionDescription
+from .key_ceremony import CoefficientValidationSet
+from .tally import CiphertextTally
+from .group import ElementModP, ElementModQ
+
+RECORD_DIR = "output" + path.sep
+COEFFICIENTS_DIR = RECORD_DIR + "coefficients" + path.sep
+BALLOTS_DIR = RECORD_DIR + "encrypted_ballots" + path.sep
+SPOILED_DIR = RECORD_DIR + "spoiled_ballots" + path.sep
+
+DESCRIPTION_FILE_NAME = "description"
+CONTEXT_FILE_NAME = "context"
+CONSTANTS_FILE_NAME = "constants"
+ENCRYPTED_TALLY_FILE_NAME = "encrypted_tally"
+TALLY_FILE_NAME = "tally"
+
+COEFFICIENT_PREFIX = "coefficient_validation_set_"
+BALLOT_PREFIX = "ballot_"
+
+
+def publish(
+    description: ElectionDescription,
+    context: CiphertextElectionContext,
+    constants: ElectionConstants,
+    coefficient_validation_sets: List[CoefficientValidationSet],
+    ciphertext_ballots: List[CiphertextBallot],
+    ciphertext_tally: CiphertextTally,
+    plaintext_tally: PlaintextTally,
+    record_directory: str = RECORD_DIR,
+) -> None:
+    """Publishes the election record as json"""
+
+    set_serializers()
+
+    make_directory(record_directory)
+    description.to_json_file(DESCRIPTION_FILE_NAME, record_directory)
+    context.to_json_file(CONTEXT_FILE_NAME, record_directory)
+    constants.to_json_file(CONSTANTS_FILE_NAME, record_directory)
+
+    make_directory(COEFFICIENTS_DIR)
+    for coefficient_validation_set in coefficient_validation_sets:
+        set_name = COEFFICIENT_PREFIX + coefficient_validation_set.owner_id
+        coefficient_validation_set.to_json_file(set_name, COEFFICIENTS_DIR)
+
+    make_directory(BALLOTS_DIR)
+    for ballot in ciphertext_ballots:
+        ballot_name = BALLOT_PREFIX + ballot.object_id
+        ballot.to_json_file(ballot_name, BALLOTS_DIR)
```

Comment:
If you've got a million ballots, which could totally happen, having a million files in a single directory is going to be a performance problem every time you try to just list the files. There needs to be some sort of subdirectory structure here.

Reply:
This is a piece we're looking at in a future card. The point is to provide a tool that individuals can use to output and a framework for this. We're more focusing on the file itself not the storage process in this PR.

Especially because developers can choose to change how these files are output.